test/onnx/test_fx_to_onnx.py

```diff
@@ -866,6 +867,26 @@ def forward(self, x):
         model = PrintModule()
         _ = torch.onnx.dynamo_export(model, input)
 
+    def test_bfloat16_serialization(self):
+        torch.manual_seed(0)
+        np.random.seed(0)
+
+        class MyLinear(nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.fc1 = nn.Linear(10, 5, bias=False, dtype=torch.bfloat16)
+
+            def forward(self, tensor_x: torch.Tensor):
+                return self.fc1(tensor_x)
+
+        model = MyLinear()
+        tensor_x = torch.rand((3, 10), dtype=torch.bfloat16)
+        onnx_program = torch.onnx.dynamo_export(model, tensor_x)
```

Comment:
Could you rebase and see if this test really works, because I think it would be blocked by https://github.com/microsoft/onnxscript/issues/1471

Also, I am not sure what is our opinion to `ExportedProgram` now, if we still think it matters (or would be main source of dynamo_export), we probably should add that.

Reply:
I see you mentioned it in the description.

Reply:
I didnt understand your point.

IIRC, using `ExportedProgram`, it works. but not using is also a supported scenario, so we should fix this one too